crates/pera-core/src/dwallet_mpc/dkg.rs

```diff
@@ -176,22 +212,17 @@ impl BytesParty for SecondDKGBytesParty {
         self,
         messages: HashMap<PartyID, Vec<u8>>,
         auxiliary_input: Vec<u8>,
-    ) -> Result<AdvanceResult, twopc_mpc::Error> {
+    ) -> PeraResult<AdvanceResult> {
         let auxiliary_input =
-            bcs::from_bytes(&auxiliary_input).map_err(|_| twopc_mpc::Error::InvalidParameters)?;
-        let messages = messages
-            .into_iter()
-            .map(|(k, v)| {
-                let message = bcs::from_bytes(&v).map_err(|_| twopc_mpc::Error::InvalidParameters);
-                return match message {
-                    Ok(message) => Ok((k, message)),
-                    Err(err) => Err(err),
-                };
-            })
-            .collect::<Result<HashMap<_, _>, _>>()?;
+            bcs::from_bytes(&auxiliary_input).map_err(|_| PeraError::DWalletMPCInvalidUserInput)?;
         let result = self
             .party
-            .advance(messages, &auxiliary_input, &mut rand_core::OsRng)?;
+            .advance(
+                deserialize_mpc_messages(messages)?,
+                &auxiliary_input,
+                &mut rand_core::OsRng,
+            )
+            .map_err(twopc_error_to_pera_error)?;
```

Comment:
Code duplication from `impl ... for FirstDKGByteParty`. Please see whether you can extract this into a function :)

Reply:
This is needed as this is the place where the actual types are being specified, and as the code duplication is only this function call it will be challenging to reduce it further